Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/preprocessor/CqlPreprocessor.kt

```diff
@@ -0,0 +1,323 @@
+        libraryBuilder.beginTranslation()
+        return try {
+            // Loop through and call visit on each child (to ensure they are tracked)
+            for (i in 0 until ctx.childCount) {
+                val tree = ctx.getChild(i)
+                val terminalNode = if (tree is TerminalNode) tree else null
+                if (terminalNode != null && terminalNode.symbol.type == cqlParser.Tokens.EOF) {
+                    continue
+                }
+                val childResult = visit(tree!!)
+                // Only set the last result if we received something useful
+                if (childResult != null) {
+                    lastResult = childResult
+                }
+            }
+
+            // Return last result (consistent with super implementation and helps w/ testing)
+            lastResult!!
+        } finally {
+            libraryBuilder.endTranslation()
+        }
+    }
+
+    override fun visitLibraryDefinition(ctx: cqlParser.LibraryDefinitionContext): Any {
+        val identifiers = visit(ctx.qualifiedIdentifier()) as MutableList<String>
+        libraryInfo.libraryName = identifiers.removeAt(identifiers.size - 1)
+        if (identifiers.size > 0) {
+            libraryInfo.namespaceName = java.lang.String.join(".", identifiers)
+        }
+        if (ctx.versionSpecifier() != null) {
+            libraryInfo.version = visit(ctx.versionSpecifier()!!) as String
+        }
+        libraryInfo.definition = ctx
+        processHeader(ctx, libraryInfo)
+        return super.visitLibraryDefinition(ctx)!!
+    }
+
+    override fun visitIncludeDefinition(ctx: cqlParser.IncludeDefinitionContext): Any {
+        val includeDefinition = IncludeDefinitionInfo()
+        val identifiers = visit(ctx.qualifiedIdentifier()) as MutableList<String>
+        includeDefinition.name = identifiers.removeAt(identifiers.size - 1)
+        if (identifiers.size > 0) {
+            includeDefinition.namespaceName = java.lang.String.join(".", identifiers)
+        }
+        if (ctx.versionSpecifier() != null) {
+            includeDefinition.version = visit(ctx.versionSpecifier()!!) as String
+        }
+        if (ctx.localIdentifier() != null) {
+            includeDefinition.localName = parseString(ctx.localIdentifier())
+        } else {
+            includeDefinition.localName = includeDefinition.name
+        }
+        includeDefinition.definition = ctx
+        processHeader(ctx, includeDefinition)
+        libraryInfo.addIncludeDefinition(includeDefinition)
+        return includeDefinition
+    }
+
+    override fun visitUsingDefinition(ctx: cqlParser.UsingDefinitionContext): Any {
+        val usingDefinition = UsingDefinitionInfo()
+        val identifiers = visit(ctx.qualifiedIdentifier()) as MutableList<String>
+        val unqualifiedIdentifier: String = identifiers.removeAt(identifiers.size - 1)
+        usingDefinition.name = unqualifiedIdentifier
+        if (identifiers.size > 0) {
```

Comment:
not empty?

Reply:
Resolved here https://github.com/cqframework/clinical_quality_language/pull/1450